### PR TITLE
Defines for flags passed to `Actor_UpdateBgCheckInfo`

### PR DESCRIPTION
**Pull request by Dragorn421**
Defines for the flags passed as the last argument to `Actor_UpdateBgCheckInfo`

Suggestions for the prefix are welcome. The current `UPDBGCHECKINFO_FLAG_` is in my opinion awful

-----------------

Side note, the combination `UPDBGCHECKINFOFLAG_0 | UPDBGCHECKINFOFLAG_2 | UPDBGCHECKINFOFLAG_3 | UPDBGCHECKINFOFLAG_4` is used 50 times out of the 232 calls to `Actor_UpdateBgCheckInfo`. It may be a good idea to add a define for this whole combination eventually if it makes particular sense

(regex `UPDBGCHECKINFOFLAG_0[\s\n]*\|[\s\n]*UPDBGCHECKINFOFLAG_2[\s\n]*\|[\s\n]*UPDBGCHECKINFOFLAG_3[\s\n]*\|[\s\n]*UPDBGCHECKINFOFLAG_4`)

### CODE REVIEW

**Review by fig02 — APPROVED** [review]
agreed on the prefix not being desired, i dont have any suggestions for now. it will prob have to wait till bgcheck flags itself is renamed.

I can see a world (using your suggestion) where they are colFlags and colInfo, in which case `COL_INFO_FLAG_` could be a possibility 

for a while now ive been saying that this could have possibly been a sub struct, all of it is grouped together. this being called `ColInfo` with the function as `UpdateColInfo` could fit pretty well 
```c
    /* 0x074 */ CollisionPoly* wallPoly; // Wall polygon the actor is touching
    /* 0x078 */ CollisionPoly* floorPoly; // Floor polygon directly below the actor
    /* 0x07C */ u8 wallBgId; // Bg ID of the wall polygon the actor is touching
    /* 0x07D */ u8 floorBgId; // Bg ID of the floor polygon directly below the actor
    /* 0x07E */ s16 wallYaw; // Y rotation of the wall polygon the actor is touching
    /* 0x080 */ f32 floorHeight; // Y position of the floor polygon directly below the actor
    /* 0x084 */ f32 yDistToWater; // Distance to the surface of active waterbox. Negative value means above water
    /* 0x088 */ u16 bgCheckFlags; // See comments below actor struct for wip docs. TODO: macros for these flags
```

**Comment by fig02** [comment]
actually the s16 that comes right after the last member here probably wouldnt align right if it were a substruct :/

File: include/z64actor.h [hunk 1]
```diff
@@ -357,4 +357,13 @@ typedef enum {
...
 } DoorLockType;
 
+#define UPDBGCHECKINFOFLAG_0 (1 << 0) // check wall
+#define UPDBGCHECKINFOFLAG_1 (1 << 1) // check ceiling
+#define UPDBGCHECKINFOFLAG_2 (1 << 2) // check floor and water
+#define UPDBGCHECKINFOFLAG_3 (1 << 3)
+#define UPDBGCHECKINFOFLAG_4 (1 << 4)
+#define UPDBGCHECKINFOFLAG_5 (1 << 5) // unused
+#define UPDBGCHECKINFOFLAG_6 (1 << 6) // disable water ripples
+#define UPDBGCHECKINFOFLAG_7 (1 << 7) // alternate wall check?
```
Comment (engineer124) [inline include/z64actor.h:367]:
```suggestion
#define UPDBGCHECKINFOFLAG_0 (1 << 0) // check wall
#define UPDBGCHECKINFOFLAG_1 (1 << 1) // check ceiling
#define UPDBGCHECKINFOFLAG_2 (1 << 2) // check floor and water
#define UPDBGCHECKINFOFLAG_3 (1 << 3)
#define UPDBGCHECKINFOFLAG_4 (1 << 4)
#define UPDBGCHECKINFOFLAG_5 (1 << 5) // unused
#define UPDBGCHECKINFOFLAG_6 (1 << 6) // disable water ripples
#define UPD_BGCHECK_INFO_FLAG_7 (1 << 7) // alternate wall check?
```
I know it's getting a bit verbose, but shouldn't these be spaced out by `_`? Seems hard to read as one big blog of text
And also should `BGCHECK_` take a higher priority than `UPD`? So would `BGCHECK_UPD_INFO_FLAG_7` make more sense?

Comment (fig02) [reply include/z64actor.h:367]: the scope of these flags is really the one function: `Actor_UpdateBgCheckInfo`
a bgcheck prefix makes it seem like its a part of bgcheck imo, and the whole "unit" or "system" is "update bgcheck info"

the name is not ideal at all as dragorn mentioned, and hopefully we can rename these variables all together to be less verbose

as far as spacing them goes, `UPDBGCHECKINFO_FLAG_7` would prob work, at least for the time being

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]
I definitely agree we need a better prefix, but I don't have a suggestion right now either

**Pull request merged**